### ISSUE TITLE
E1136: Value 'pydantic.Json' is unsubscriptable (unsubscriptable-object)

### ISSUE THREAD

**Issue by evgenymarkov** [fcfangcc/pylint-pydantic#11]
Hello ✋ I found that Pylint incorrectly reports that pydantic.Json type is unsubscriptable object.

Example:

```python
import pydantic

RolesForUnits = dict[str, str]


class Settings(pydantic.BaseSettings):
    ROLES_FOR_UNITS: pydantic.Json[RolesForUnits] = {}

    class Config(pydantic.BaseSettings.Config):
        case_sensitive = True


__all__ = [
    "Settings",
]
```

Pylint report:

```
E1136: Value 'pydantic.Json' is unsubscriptable (unsubscriptable-object)
```

How type is defined in the Pydantic source code:

```python
Json = Annotated[T, ...]
```

**Comment by fcfangcc** [comment]
I can't repeat it. Please tell me

* pylint version: 
* pydatic version: 
* python version:
* pylint-pydantic version:
* other more: ...

**Comment by evgenymarkov** [comment]
No problem, I made repository with example – https://github.com/evgenymarkov/pylint-pydantic-json

Python: 3.10.6
Pydantic: 1.10.4
Pylint: 2.15.10
Pydantic plugin: 0.1.5

**Comment by fcfangcc** [comment]
maybe your repository with example is private ? I open it then show 404.
